.evergreen/config_generator/components/std/std_c11.py

```diff
@@ -0,0 +1,52 @@
+from shrub.v3.evg_build_variant import BuildVariant
+from shrub.v3.evg_task import EvgTaskRef
+
+from config_generator.etc.compile import generate_compile_tasks
+
+from config_generator.components.sasl.openssl import SaslAutoOpenSSLCompile
+
+
+TAG = 'std-matrix-c11'
+
+
+# pylint: disable=line-too-long
+# fmt: off
+COMPILE_MATRIX = [
+    ('archlinux',  'clang', None,   ['auto']),
+    ('debian81',   'clang', None,   ['auto']),
+    ('debian92',   'clang', None,   ['auto']),
+    ('ubuntu1604', 'clang', 'i686', ['auto']),
+    ('ubuntu1604', 'clang', None,   ['auto']),
+    ('ubuntu1804', 'clang', 'i686', ['auto']),
+    ('ubuntu1804', 'gcc',   None,   ['auto']),
+]
```

Comment:
Given that even recent compiler versions have made substantial improvements in c11 support, I think it would make sense to add ubuntu2004 and debian10/debian11 to the matrix.

```suggestion
    ('debian10',   'clang', None,   ['auto']),
    ('debian10',   'gcc',   None,   ['auto']),
    ('debian11',   'clang', None,   ['auto']),
    ('debian11',   'gcc',   None,   ['auto']),
    ('ubuntu2004', 'clang', None,   ['auto']),
    ('ubuntu2004', 'gcc',   None,   ['auto']),
]
```

Also, the gcc in both Debian 10 and 11 supports c17.  Do we need to think about adding a configuration to test with `-std=c17`?